server/utils/commonFunc.go

```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	"errors"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+func IsPassValid(password string) error {
+
+	if len(password) < 8 {
+		return errors.New("password is too short")
+
+	}
+	hasUpperCase := false
+	hasLowerCase := false
+	hasNumbers := false
+	hasSpecial := false
+
+	for _, char := range password {
+		if char >= 'A' && char <= 'Z' {
+			hasUpperCase = true
+		} else if char >= 'a' && char <= 'z' {
+			hasLowerCase = true
+		} else if char >= '0' && char <= '9' {
+			hasNumbers = true
+		} else if char >= '!' && char <= '/' {
+			hasSpecial = true
+		} else if char >= ':' && char <= '@' {
+			hasSpecial = true
+		}
+	}
+
+	if !hasUpperCase {
+		return errors.New("password do not contain upperCase Character")
+	}
+
+	if !hasLowerCase {
+		return errors.New("password do not contain LowerCase Character")
+	}
+
+	if !hasNumbers {
+		return errors.New("password do not contain any numbers")
+	}
+
+	if !hasSpecial {
+		return errors.New("password do not contain any special character")
+	}
+	return nil
+}
+
+func HashPassword(password string) (*string, error) {
+	bs, err := bcrypt.GenerateFromPassword([]byte(password), 10)
```

Comment:
what is `10` in bcrypt.GenerateFromPassword([]byte(password), 10)

Reply:
Sir, it is the cost for generating the hash. Default is 10 so i have used 10